### DISCUSSION TITLE
How are "failing" test folders used/managed?

### DISCUSSION THREAD

**Discussion by yihtserns** [FasterXML/jackson-databind#4642] [General]
Right now tests in `failing` folders are not executed when running `mvn test`, so how would you know when they're no longer failing?

Do the maintainers periodically check them?  Or you search for the issue number for any failing test when working on the issue?  What's the process/workflow involving that folder?

---
For example, some test methods are passing, are the maintainers aware of this but leaving it in `failing` folder since they're not "officially"/"properly" supported so are considered passing by chance/coincidence/accidentally:

![image](https://github.com/user-attachments/assets/f34d9250-f289-414d-961e-053117e3e1a6)

**Comment by pjfanning** [comment]
> Right now tests in `failing` folders are not executed when running `mvn test`, so how would you know when they're no longer failing?
> 
> Do the maintainers periodically check them? What's the process/workflow involving that folder?

You can run them in an IDE.

**Reply by yihtserns** [reply]
Does that mean maintainers periodically run all the tests in the IDE, and check if any happens to pass?

**Reply by yihtserns** [reply]
Or maybe I wasn't being clear: wasn't asking how to run them, but how they're being maintained/managed.

**Reply by pjfanning** [reply]
If you find any that pass, can you do a PR and move the test out of failing?

Most of the tests that are in failing are related to issues and we should track down the related issues to update them.

**Reply by pjfanning** [reply]
@yihtserns I created #4643 but there are maybe 3 other tests elsewhere that can be enabled

**Reply by pjfanning** [reply]
also now #4644

**Reply by pjfanning** [reply]
adding #4645

**Reply by pjfanning** [reply]
I think #4646 now covers all 4 working tests

Thanks @yihtserns for highlighting this

**Reply by yihtserns** [reply]
OK while @cowtowncoder clarified that those above shouldn't be done (@pjfanning sorry about that, and thanks for the effort...), I found one that is definitely good to go - this disabled test is passing: https://github.com/FasterXML/jackson-databind/blob/aafba8790fad349095b9e59ee5082386f34d601e/src/test/java/com/fasterxml/jackson/databind/deser/creators/TestCreators.java#L423-L434

**Reply by cowtowncoder** [reply]
Ahh. Yes, those should definitely be changed, I'll find and change it.

And as to other split cases: I am not dead set against inclusion of some of passing cases. But should be done on case-by-case basis.

**Reply by cowtowncoder** [reply]
Ok there's just one such test. Thanks @yihtserns !

**Comment by cowtowncoder** [comment]
Ok, so as for me: I ran these every now and then, to see if some tests were fixed (once every 2 weeks perhaps); plus whenever working in area (or merging PRs in area) where I recall related failing tests.

And yes: if full test classes start passing, they should be moved to their regular place so `../failing/` should only contain test classes with at least 1 failure.
NOTE: it's perfectly fine for some test methods in classes to pass; these are usually to give contrast/context to failures (this case passes while that doesn't). This is not accidental.
But if and how to include passing cases as part of failing tests is judgment call.

**Comment by cowtowncoder** [comment]
@pjfanning @yihtserns There is a misunderstanding here, indeed -- those passing cases can remain there. The only hard rule is if the whole test class passes (all methods/cases pass), then class should be moved out (and possibly merged).
There is no requirement that none of the cases pass.

**Comment by yihtserns** [comment]
The reason I'm asking this is because I wanted to come up with a mechanism that I thought would make managing failing test cases more "automated" and needed less "human touch".  But from the replies above, seems like the current process/workflow can't be simplistically automated away.  

Since I've already spent time on it, I decided to just share what I've created: #4648 - maybe it can be a seed that leads to a more fit-for-purpose idea in the future

Thanks for taking the time to answer my questions @pjfanning @cowtowncoder.  🤝
